### PR TITLE
feat(sdk): add `Bucket.metadata()` for `tf-gcp` target

### LINKED ISSUE #4332: Implement `cloud.Bucket` inflight method `metadata` for GCP target
```js
/**
 * Get the metadata of an object in the bucket.
 */
inflight metadata(key: str): ObjectMetadata;

/**
 * Metadata of a bucket object.
 */
struct ObjectMetadata {
  /** The size of the object in bytes. */
  size: num;
  /** The time the object was last modified. */
  lastModified: Datetime;
  /** The content type of the object, if it is known. */
  contentType: str?;
}
```

### PR DESCRIPTION
**Pull request by garysassano**
Closes #4332

## Checklist

- [x] Title matches [Winglang's style guide](https://www.winglang.io/contributing/start-here/pull_requests#how-are-pull-request-titles-formatted)
- [x] Description explains motivation and solution
- [x] Tests added (always)
- [ ] Docs updated (only required for features)
- [ ] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Wing Cloud Contribution License](https://github.com/winglang/wing/blob/main/CONTRIBUTION_LICENSE.md)*.

### CODE REVIEW

**Review by subh-cs — COMMENTED** [review]
Great work!! Did you test this on gcp?

**Comment by garysassano** [comment]
@subh-cs Didn't test it but it should work since the code was taken straight from the [docs examples](https://googleapis.dev/nodejs/storage/latest/File.html#getMetadata-examples).

**Comment by tsuf239** [comment]
> @subh-cs Didn't test it but it should work since the code was taken straight from the [docs examples](https://googleapis.dev/nodejs/storage/latest/File.html#getMetadata-examples).

please test :)

**Comment by garysassano** [comment]
I've just deployed this to GCP and it works as expected.

**Comment by garysassano** [comment]
@tsuf239 The `Bucket.metadata()` method works as intended. The problem is with `Bucket.put()` and `Bucket.putJson()` not setting the right Content-Type when uploading. I will fix those later.

gcs-files

**Comment by tsuf239** [comment]
I promise to review it right after adding the GCP test option

**Comment by garysassano** [comment]
Keep.

File: examples/tests/sdk_tests/bucket/metadata.test.w [hunk 1]
```diff
@@ -1,36 +1,57 @@
...
-
-  let file1Metadata = b.metadata("file1.main.w");
-  assert(file1Metadata.size == 3);
-  assert(file1Metadata.contentType == "application/octet-stream");
+  let assertThrows = (expected: str, block: (): void) => {
+    let var error = false;
+    try {
+      block();
+    } catch actual {
+      assert(actual.contains(expected));
```
Comment (tsuf239) [inline examples/tests/sdk_tests/bucket/metadata.test.w:12]: why is it checking for "contains" and not equals?

**Review by tsuf239 — APPROVED** [review]
Awesome job! ✨ ✨ 
Working on both tf-aws and tf-gcp:
image
image

File: examples/tests/sdk_tests/bucket/metadata.test.w (hunk 1 above)
Comment (garysassano) [reply examples/tests/sdk_tests/bucket/metadata.test.w:12]: Same reason explained [here](https://github.com/winglang/wing/pull/4999#discussion_r1427456384).

**Pull request merged**